engine/worker/cmd_cache.go

```diff
@@ -0,0 +1,283 @@
+	if errRead != nil {
+		errRead = sdk.WrapError(errRead, "Cannot read body")
+		writeError(w, r, errRead)
+		return
+	}
+
+	var c sdk.Cache
+	if err := json.Unmarshal(data, &c); err != nil {
+		err = sdk.WrapError(err, "Cannot unmarshall body")
+		writeError(w, r, err)
+		return
+	}
+	sendLog := getLogger(wk, wk.currentJob.wJob.ID, wk.currentJob.currentStep)
+
+	res, errTar := sdk.CreateTarFromPaths(c.Files)
+	if errTar != nil {
+		errTar = sdk.WrapError(errTar, "worker cache push > Cannot tar")
+		sendLog(errTar.Error())
+		writeError(w, r, errTar)
+		return
+	}
+	if wk.currentJob.wJob == nil {
+		errW := fmt.Errorf("worker cache push > Cannot find workflow job info")
+		sendLog(errW.Error())
+		writeError(w, r, errW)
+		return
+	}
+	params := wk.currentJob.wJob.Parameters
+	projectKey := sdk.ParameterValue(params, "cds.project")
+	if err := wk.client.WorkflowCachePush(projectKey, vars["tag"], res); err != nil {
+		err = sdk.WrapError(err, "worker cache push > Cannot push cache")
+		sendLog(err.Error())
+		writeError(w, r, err)
+		return
+	}
+}
+
+func cmdCachePull(w *currentWorker) *cobra.Command {
+	c := &cobra.Command{
+		Use:     "pull",
+		Aliases: []string{"download"},
+		Short:   "worker cache pull tagValue",
+		Long: `
+Inside a project, you can fetch a cache from your worker with a tag
+		`,
+		Run: cachePullCmd(w),
+	}
+	return c
+}
+
+func cachePullCmd(w *currentWorker) func(cmd *cobra.Command, args []string) {
+	return func(cmd *cobra.Command, args []string) {
+		portS := os.Getenv(WorkerServerPort)
+		if portS == "" {
+			sdk.Exit("worker cache pull > %s not found, are you running inside a CDS worker job?\n", WorkerServerPort)
+		}
+
+		port, errPort := strconv.Atoi(portS)
+		if errPort != nil {
+			sdk.Exit("worker cache pull > cannot parse '%s' as a port number", portS)
+		}
+
+		if len(args) < 1 {
+			sdk.Exit("worker cache pull > Wrong usage: Example : worker cache pull myTagValue")
+		}
+
+		fmt.Printf("Worker cache pull in progress... (tag: %s)\n", args[0])
+		req, errRequest := http.NewRequest("GET", fmt.Sprintf("http://127.0.0.1:%d/cache/%s/pull", port, args[0]), nil)
+		if errRequest != nil {
+			sdk.Exit("worker cache pull > cannot post worker cache pull (Request): %s\n", errRequest)
+		}
+
+		client := http.DefaultClient
+		client.Timeout = 10 * time.Minute
+
+		resp, errDo := client.Do(req)
+		if errDo != nil {
+			sdk.Exit("worker cache pull > cannot post worker cache pull (Do): %s\n", errDo)
+		}
+
+		if resp.StatusCode >= 300 {
+			sdk.Exit("worker cache pull > Cannot cache pull HTTP ERROR %d\n", resp.StatusCode)
+		}
+
+		fmt.Printf("Worker cache pull with success (tag: %s)\n", args[0])
+	}
+}
+
+func (wk *currentWorker) cachePullHandler(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+
+	sendLog := getLogger(wk, wk.currentJob.wJob.ID, wk.currentJob.currentStep)
+
+	if wk.currentJob.wJob == nil {
+		errW := fmt.Errorf("worker cache pull > Cannot find workflow job info")
+		sendLog(errW.Error())
```

Comment:
useless to send error, it will be displayed twice if you check the error in the command